internal/commands/user/delete_test.go

```diff
@@ -0,0 +1,309 @@
+package user
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestUserDeleteSetup(t *testing.T) {
+	t.Run("should construct a realm client with the configured base url", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		cmd := &CommandDelete{inputs: deleteInputs{}}
+		assert.Nil(t, cmd.realmClient)
+
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+	})
+}
+
+func TestUserDeleteHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	testApp := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+	testUsers := []realm.User{
+		{
+			ID: "user-1",
+			Identities: []realm.UserIdentity{
+				{ProviderType: providerTypeAnonymous},
+			},
+			Disabled: false,
+		},
+	}
+
+	t.Run("should delete a user when a user id is provided", func(t *testing.T) {
+		var capturedAppFilter realm.AppFilter
+		var capturedFindProjectID, capturedFindAppID string
+		var capturedDeleteProjectID, capturedDeleteAppID string
+
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			capturedAppFilter = filter
+			return []realm.App{testApp}, nil
+		}
+
+		realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+			capturedFindProjectID = groupID
+			capturedFindAppID = appID
+			return testUsers, nil
+		}
+
+		realmClient.DeleteUserFn = func(groupID, appID, userID string) error {
+			capturedDeleteProjectID = groupID
+			capturedDeleteAppID = appID
+			return nil
+		}
+
+		cmd := &CommandDelete{
+			inputs: deleteInputs{
+				ProjectInputs: app.ProjectInputs{
+					Project: projectID,
+					App:     appID,
+				},
+				usersInputs: usersInputs{
+					Users: []string{testUsers[0].ID},
+				},
+			},
+			realmClient: realmClient,
+		}
+
+		assert.Nil(t, cmd.Handler(nil, nil))
+		assert.Equal(t, realm.AppFilter{App: appID, GroupID: projectID}, capturedAppFilter)
+		assert.Equal(t, projectID, capturedFindProjectID)
+		assert.Equal(t, appID, capturedFindAppID)
+		assert.Equal(t, projectID, capturedDeleteProjectID)
+		assert.Equal(t, appID, capturedDeleteAppID)
+		assert.Equal(t, testUsers[0].ID, cmd.inputs.Users[0])
+		assert.Nil(t, cmd.outputs[0].err)
+		assert.Equal(t, cmd.outputs[0].user, testUsers[0])
+	})
+
+	t.Run("should save failed deletion errors", func(t *testing.T) {
```

Comment:
can we test table these two tests?  I think you could achieve this pretty easily with

```go
struct{
  description string
  userDeleteErr error
  expectedOutputs []userOutput
}
```

also, use the asserts from the test above, they are a bit more specific